testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppWindow.kt

```diff
@@ -1,25 +1,40 @@
-package com.rakuten.tech.mobile.testapp.ui.display.firsttime
+package com.rakuten.tech.mobile.testapp.ui.display.preload
 
 import android.annotation.SuppressLint
 import android.content.Context
 import android.content.SharedPreferences
 import android.net.Uri
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.widget.TextView
+import android.widget.Toast
 import androidx.appcompat.app.AlertDialog
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.rakuten.tech.mobile.miniapp.MiniApp
 import com.rakuten.tech.mobile.miniapp.MiniAppInfo
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionResult
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionType
 import com.rakuten.tech.mobile.miniapp.testapp.R
 import com.rakuten.tech.mobile.testapp.helper.setIcon
+import com.rakuten.tech.mobile.testapp.helper.showAlertDialog
+import com.rakuten.tech.mobile.testapp.ui.settings.AppSettings
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
 import java.lang.Exception
 
 private const val DEFAULT_ACCEPTANCE = false
 
 class PreloadMiniAppWindow(private val context: Context, private val preloadMiniAppLaunchListener: PreloadMiniAppLaunchListener) {
+    private val miniApp = MiniApp.instance(AppSettings.instance.miniAppSettings)
     private lateinit var preloadMiniAppAlertDialog: AlertDialog
     private lateinit var preloadMiniAppLayout: View
     private var miniAppInfo: MiniAppInfo? = null
     private var miniAppId: String = ""
+    private var miniAppVersionId: String = ""
```

Comment:
version id is empty when trying to download a miniapp using only app-id from this page. 

<img width="308" alt="Screen Shot 2021-02-15 at 14 21 05" src="https://user-images.githubusercontent.com/66930373/107908312-1486f380-6f99-11eb-8161-bf6458fb0c26.png">

Then metadata's permission value will not be found in first time UI. 
@corycaywood Do you think, this will be expected?

Reply:
This one is supposed to launch the latest version ID for the specified mini app. So I think it needs to first call `MiniApp.fetchInfo` to get the latest version ID and then fetch the meta data for that version.